tree_node.go

```diff
@@ -139,11 +134,14 @@ func (node *TreeNode) Copy() *TreeNode {
 		path:         node.path,
 		depth:        node.depth,
 		hasher:       node.hasher,
-		extended:     node.extended,
+		temporary:    node.temporary,
 	}
 }
 
 func (node *TreeNode) Prune(oldestVersion Version) {
+	// If the Child Node has not been updated for a long time, it will be released from memory
+	defer node.release(oldestVersion)
```

Comment:
I think the version before `oldestVersion` can hardly say long enough, the actually node in memory will be quite few.

Reply:
modified, please check the latest code